### DISCUSSION TITLE
Enhanced `ElSelect`, supports multiple selections with string binding.  增强 Select，支持多选时绑定值为字符串，投票我将实现它 🌟

### DISCUSSION THREAD

**Discussion by warmthsea** [element-plus/element-plus#17169] [Polls]
### 为什么 Why:

- 现在的 `ElSelect`，多选的数据格式**只限数组**：`['1', '2']` Now `ElSelect`, multiple only supports binding Array：`['1', '2']` 

``` vue
<script lang="ts" setup>
 const value = ref<string[]>([])
</script> 

<template>
  <el-select v-model="value" multiple>
    <el-option label="option 1" value="1" />
    <el-option label="option 2" value="2" />
  </el-select>
</tempalte>
```

- 在具体业务中，服务端**要求的数据格式为以逗号分隔的字符串** ：`1,2` In specific business, the data format required by the server is a comma-separated string：`1,2` 
 
  - 前端需要转两次数据结构：
    - 1️⃣ 给服务端时：`['1', '2']` → `1,2`
    - 2️⃣ 回显时：`1,2` → `['1', '2']`
   
``` vue
<script lang="ts" setup>
const value = ref<string[]>([])

funciton getData() {
  getApi().then(res =>{
    const data = res.data  // 1,2
    value.value = data.split(',') // → `['1', '2']`
  })
}

funciton setData() {
  const data = value.value.join(',') // → `1,2`
  setApi(data).then(res =>{
     console.log('save success!')
  })
}
</script> 

<template>
  <el-select v-model="value" multiple>
    <el-option label="option 1" value="1" />
    <el-option label="option 2" value="2" />
  </el-select>
  <el-button @click="setData()">submit form</el-button>
</tempalte>
```

上面只是一个简单的示例，在业务功能复杂的页面可能存在不同的深层数据结构内存在多个多选框，处理这些数据变的非常繁琐。

如果 Select 支持字符串多选，将有以下效果：

![screenshots](https://github.com/element-plus/element-plus/assets/45450994/ee9478a7-fda7-4cc9-8325-dc29931d48a4)

**Comment by xxholly32** [comment]
我感觉做进组件库不太合适，你说的这种情况在业务中确实会比较多，但其实也不止你说的情况，可能会更加复杂。

从我个人的角度的话可能说是业务中处理，或者可以使用一个composable去减轻业务的压力。

**Comment by jinceon** [comment]
前端提交给服务器端的时候 自己转下
或者让服务器改成数组
